src/cobra/core/gene.py

```diff
@@ -147,7 +129,16 @@ def parse_gpr(str_expr):
     -------
     tuple
         elements ast_tree and gene_ids as a set
+
```

Comment:
Again code duplication, this should use `return GPR(str_expr).body`. This way the tests actually use the new implementation and we can check whether it works in all edge cases.

Reply:
It's not fully backwards compatible (old one returned the AST tree not a GPR object) but I would personally be fine with that since the actual usage does not change.